reliability-engineering/terraform/modules/concourse-worker-pool/files/worker-init.sh

```diff
@@ -63,14 +63,56 @@ LimitNOFILE=infinity
 WantedBy=multi-user.target
 EOF
 
+# adapted from
+# https://aws.amazon.com/blogs/compute/best-practices-for-handling-ec2-spot-instance-interruptions/
+# 'EOF' in single quotes prevents varible interpolation; this is important
+cat <<'EOF' > /usr/local/bin/check-spot-interruption
+#!/bin/bash
+set -ue
+TOKEN=`curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 1800"`
+
+while sleep 5; do
+
+    HTTP_CODE=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" -s -w %{http_code} -o /dev/null http://169.254.169.254/latest/meta-data/spot/instance-action)
+
+    if [[ "$HTTP_CODE" -eq 401 ]] ; then
+        echo 'Refreshing Authentication Token'
+        TOKEN=`curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 1800"`
+    elif [[ "$HTTP_CODE" -eq 200 ]] ; then
+        echo 'Interrupted: retiring concourse-worker'
+        systemd kill -s SIGUSR2 concourse-worker
```

Comment:
there's also `concourse retire-worker` comand ... I'm not sure if it does anything that direct SIGUSR2 doesn't do.

Reply:
as far as I can tell from the docs:

 - `SIGUSR2` and `concourse retire-worker` are the same thing
 - retiring and landing are separate. [this thread suggests you can't retire a landed worker](https://discuss.concourse-ci.org/t/whats-the-grace-way-to-remove-a-worker/897/4?u=philandstuff) (which shows the problem of metaphors, because I'd hate to be on a plane that retired mid flight 😄 )